core/Inst.hpp

```diff
@@ -40,6 +40,7 @@ namespace olympia
                 uint32_t val = 0;
                 core_types::RegFile rf = core_types::RegFile::RF_INVALID;
                 mavis::InstMetaData::OperandFieldID field_id = mavis::InstMetaData::OperandFieldID::NONE;
+                bool x0 = false;
```

Comment:
```
struct Reg {
    const uint32_t val;
    const core_types::RegFile rf;
    const mavis::InstMetaData::OperandFieldID field_id;
    const bool is_x0;

    Reg(uint32_t v, core_types::RegFile rf, mavis::InstMetaData::OperandFieldID field_id, bool is_x0)
        : val(v), rf(rf), field_id(field_id), is_x0(is_x0) {}
};
```

So I attempted to try and make the struct data members constant, but we then run into issues with the getter/setter functions because the copy constructor for the struct is deleted due to `const` on the data members.

Reply:
Ah, right.  I forgot you're replacing the internal members with new values.  Yeah, adding `const` will make the `Reg` struct ill-formed for assignment or move assignment.  Oh well.